ui.frontend/src/view/FormFieldBase.js

```diff
@@ -187,15 +187,16 @@ class FormFieldBase extends FormField {
     setFocus(id) {
         const fieldType = this.parentView?.getModel()?.fieldType;
         if (fieldType !== 'form' && this.parentView.setFocus) {
-            this.parentView.setFocus(id);
+            this.parentView.setFocus(this.getId());
         }
-        // If multiple widgets like radio-button or checkbox-group, then focus on the first widget
-        if (this.widget.length > 0) {
-            this.widget[0].focus();
+        this.widget = this.getWidget(); // updating to the latest widget in case of datepicker widget with a formatter
+        if (this.widget instanceof NodeList) {
+            this.widget[0].focus(); // If multiple widgets like radio-button or checkbox-group, then focus on the first widget
+        } else if(this.widget.type === 'file') {
```

Comment:
You should rather check the base class here instead of relying on widget type. Rajat has already created a base class for file input.